packages/trader/src/AppV2/Containers/Chart/trade-chart.tsx

```diff
@@ -29,12 +26,14 @@ const BottomWidgetsMobile = ({ digits, tick, setTickData, setDigitStats }: TBott
 
     React.useEffect(() => {
         setDigitStats(digits);
+        // For digits array, which is coming from SmartChart, reference is not always changing.
+        // As it is the same, this useEffect was not triggered on every array update.
         // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [digits]);
+    }, [digits.join('-')]);
```

Comment:
nice logic

Reply:
not asking to change , this is just a question

Reply:
I was afraid to add changes to Charts + it will require a huge sanity(DTrader, DBots?) while our QAs have really a lot cards currently. It would be nice to refactor Charts, but I think it might be better to do it during chart revamp (when we will be implementing new charts for DTrader-V2)🙇🏻‍♀️